Modules/_struct.c

```diff
@@ -1669,9 +1669,15 @@ s_unpack_internal(PyStructObject *soself, const char *startfrom,
             if (e->format == 's') {
                 v = PyBytes_FromStringAndSize(res, code->size);
             } else if (e->format == 'p') {
-                Py_ssize_t n = *(unsigned char*)res;
-                if (n >= code->size)
-                    n = code->size - 1;
+                Py_ssize_t n;
+                if (code->size == 0) {
+                    n = 0;
+                } else {
```

Comment:
[PEP7](https://peps.python.org/pep-0007/) ;)
```suggestion
                }
                else {
```

Reply:
Thanks!